packages/good-design/src/apps/bridge/MicroBridge.tsx

```diff
@@ -0,0 +1,232 @@
+  };
+  fees?: { [key: string]: { minFee: BigNumber; maxFee: BigNumber; fee: BigNumber } };
+  bridgeStatus?: Partial<TransactionStatus>;
+  relayStatus?: Partial<TransactionStatus>;
+  selfRelayStatus?: Partial<TransactionStatus>;
+}) => {
+  const [isBridging, setBridging] = useState(false);
+  const [inputWei, setInput] = useState<string>("0");
+  const expectedFee = Number((Number(inputWei) * 0.001) / 100).toFixed(2);
+  const [enableTarget, setEnableTarget] = useState(false);
+
+  const [target, setTarget] = useState<string | undefined>();
+
+  const [sourceChain, setSourceChain] = useState<"fuse" | "celo">("fuse");
+
+  const targetChain = sourceChain === "fuse" ? "celo" : "fuse";
+  const balanceWei = useBalanceHook(sourceChain);
+
+  let { isValid, reason } = useCanBridge(sourceChain, inputWei);
+  const hasBalance = Number(inputWei) <= Number(balanceWei);
+  const isTargetValid = !enableTarget || isAddressValid(target || "");
+  const isValidInput = isValid && hasBalance && isTargetValid;
+  reason = reason || (!hasBalance && "balance") || (!isTargetValid && "target") || "";
+
+  const toggleChains = useCallback(() => {
+    setSourceChain(targetChain);
+    onSetChain && onSetChain(targetChain);
+  }, [setSourceChain, onSetChain, targetChain]);
+
+  const triggerBridge = useCallback(async () => {
+    setBridging(true);
+    try {
+      await onBridge(inputWei, sourceChain, enableTarget ? target : undefined);
+    } catch (e) {
+      setBridging(false);
+    }
+  }, [setBridging, onBridge, inputWei, sourceChain, enableTarget, target]);
+
+  useEffect(() => {
+    if (
+      ["Mining", "PendingSignature", "Success"].includes(bridgeStatus?.status || "") &&
+      (["None", "Fail", "Exception", "Success"].includes(selfRelayStatus?.status || "") &&
+        ["None", "Fail", "Exception", "Success"].includes(relayStatus?.status || "")) === false
+    ) {
+      setBridging(true);
+    } else {
+      setBridging(false);
+    }
+  }, [relayStatus, bridgeStatus, selfRelayStatus]);
+
+  return (
+    <Box>
+      <Flex direction="row" justifyContent={"space-between"}>
+        <Box>
+          <Text textTransform="capitalize">{sourceChain}</Text>
+        </Box>
+        <Switch boxSize="8" size="lg" isChecked={sourceChain === "fuse"} onToggle={toggleChains} />
+        <Box>
+          <Text textTransform="capitalize">{sourceChain === "fuse" ? "Celo" : "Fuse"}</Text>
+        </Box>
+      </Flex>
+      <TokenInput
+        balanceWei={balanceWei}
+        decimals={2}
+        onChange={setInput}
+        minAmountWei={limits?.[sourceChain]?.minAmount?.toString()}
+      />
+      <FormControl isInvalid={!!reason}>
+        <FormControl.ErrorMessage leftIcon={<WarningOutlineIcon variant="outline" />}>
+          {reason}
+        </FormControl.ErrorMessage>
+      </FormControl>
+      <Box mt="5">
+        <Checkbox isChecked={enableTarget} value="" onChange={() => setEnableTarget(!enableTarget)}>
```

Comment:
1. inline callbacks are bad for optimization
2. useState setter support accepting callback (like this.setState in class component) so you don;'t need to reference state var
3. everything could be done with the simple useCallback will never change:

```
const toggleEnableTarget = useCallback(() => setEnableTarget(value => !value), [setEnableTarget])

// ...
onChange={toggleEnableTarget}
```